### PR TITLE
[Go-client] Fix inability to download files

### PR DESCRIPTION
**Pull request by icubbon**
Revert a change to the Go Client code that broke the ability to download files. This reverts the change made during the switch go-experimental
from https://github.com/OpenAPITools/openapi-generator/commit/14d41310b968b47bf2a6bc89bbc9c7dadffaca06

Fix for https://github.com/OpenAPITools/openapi-generator/issues/8039
The decoding for type `file` had been removed when the generator was switched to `go-experimental`
Re-added the code that existed in the 4.x release to handle downloading a file.

Tested by generating the pet-store examples to ensure that the previous code for downloading a file was present.
Generated an internal Go Rest API client that downloads a file from a Rest API. Then downloaded a file successfully.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvn clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@antihax @grokify @kemokemo @jirikuncar @ph4r5h4d

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/go/client.mustache [hunk 1]
```diff
@@ -419,6 +419,15 @@ func (c *APIClient) decode(v interface{}, b []byte, contentType string) (err err
 		*s = string(b)
 		return nil
 	}
+	if f, ok := v.(**os.File); ok {
+		*f, err = ioutil.TempFile("", "HttpClientFile")
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/go/client.mustache:423]: I got the following errors when testing the petstore go client: 
```
go-petstore/client.go:391:13: undefined: ioutil
```

Can you please take a look when you've time?

Comment (icubbon) [reply modules/openapi-generator/src/main/resources/go/client.mustache:424]: Sorry about that. That is my mistake for missing that as my dev environment will do auto-imports that hid this.

Comment (wing328) [reply modules/openapi-generator/src/main/resources/go/client.mustache:424]: Looks like you merge another PR into your PR. 

Can you file a PR with just the fix for downloading the files?

**Comment by wing328** [comment]
I've cherry-pick your commits and filed https://github.com/OpenAPITools/openapi-generator/pull/9812 instead. Closing this one.

**Pull request closed without merging**